src/WebJobs.Script/Workers/Rpc/Configuration/DynamicWorkerConfigurationResolver.cs

```diff
@@ -0,0 +1,278 @@
+            {
+                var versionDir = Path.GetFileName(workerVersionPath);
+
+                if (Version.TryParse(versionDir, out Version version))
+                {
+                    versionPathMap[version] = workerVersionPath;
+                }
+                else
+                {
+                    _logger.LogDebug("Failed to parse worker version '{versionDir}' as a valid version.", versionDir);
+                }
+            }
+
+            return versionPathMap;
+        }
+
+        /// <summary>
+        /// Determines if the worker is compatible with Host by checking if Host satisfies worker requirements.
+        /// </summary>
+        private bool IsWorkerCompatibleWithHost(string workerDirPath, JsonElement workerConfigJson)
+        {
+            if (workerConfigJson.TryGetProperty(RpcWorkerConstants.HostRequirementsSectionName, out JsonElement hostRequirementsSection))
+            {
+                _logger.LogDebug("Worker configuration at '{workerDirPath}' specifies host requirements {requirements}.", workerDirPath, hostRequirementsSection);
+
+                var hostRequirements = hostRequirementsSection.Deserialize<HashSet<string>>(JsonSerializerOptionsProvider.CaseInsensitiveJsonSerializerOptions);
+
+                if (hostRequirements is not null && !hostRequirements.IsSubsetOf(ScriptConstants.HostCapabilities))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Checks if the provided probing path is valid by ensuring it is not null and the directory exists in the file system.
+        /// </summary>
+        private bool IsValidProbingPath(string probingPath)
+        {
+            if (string.IsNullOrWhiteSpace(probingPath))
+            {
+                return false;
+            }
+
+            if (!_fileSystem.Directory.Exists(probingPath))
+            {
+                _logger.LogDebug("Worker probing path directory does not exist: {probingPath}.", probingPath);
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Determines if the current release channel is either the standard or extended platform channel.
+        /// </summary>
+        private bool IsStandardOrExtendedChannel()
+        {
+            var releaseChannel = _resolverOptions.CurrentValue.ReleaseChannel;
+
+            return !string.IsNullOrWhiteSpace(releaseChannel) &&
+                    (releaseChannel.Equals(ScriptConstants.StandardPlatformChannelNameUpper, StringComparison.OrdinalIgnoreCase) ||
+                    releaseChannel.Equals(ScriptConstants.ExtendedPlatformChannelNameUpper, StringComparison.OrdinalIgnoreCase));
+        }
+
+        /// <summary>
+        /// Comparer for sorting Version objects in descending order.
+        /// </summary>
+        private class DescendingVersionComparer : IComparer<Version>
```

Comment:
the tiniest of performance optimizations, but a good habit to have. When you have these one-off implementations that are effectively static, make a re-usable singleton instance for them:

``` CSharp
private class DescendingVersionComparer : IComparer<Version>
{
   public static readonly DescendingVersionComparer Instance = new();
   // ....
}
```